### ISSUE TITLE
Native Apple Silicon support

### ISSUE THREAD

**Issue by KovalevArtem** [kyleneideck/BackgroundMusic#395]
Add full support for new processors Apple Silicon

**Comment by ridicolos** [comment]
Are there any updates to this? I would love to see a native version vor the M1 chips, because I don't like it to be running under rosetta in the background.

**Comment by kyleneideck** [comment]
@ridicolos Have you tried the latest snapshot build [0.4.0-SNAPSHOT-b38f6dd](https://github.com/kyleneideck/BackgroundMusic/releases/tag/0.4.0-SNAPSHOT-b38f6dd)? It has universal binaries, so should run natively on Apple Silicon. I think I tested that at some point, but not 100% sure.

**Comment by ridicolos** [comment]
@kyleneideck I tried to install the version, but got an error. The old version, which I installed with homebrew, was removed with

```bash
cd /Applications/Background\ Music.app/Contents/Resources/
bash uninstall.sh
brew remove background-music --cask
``` 

if I try to install the latest snapshot build you suggested it always fails after 10 seconds of the installation. I have the lastest version of macOS (11.5.2) installed.


Also, when will this be available in the normal version so tools like homebrew will download the universal binaries?

**Comment by kyleneideck** [comment]
@ridicolos Just a generic "install failed" error? Does it log anything interesting in `/var/log/install.log`?

I haven't installed 11.5 yet, so that could be it. New major versions of the OS seem to always break Background Music in some way.

Can you try extracting the files from the .pkg and installing them manually to see if that helps narrow down the problem? This should work:

```
pkgutil --expand-full BackgroundMusic-0.4.0-SNAPSHOT-b38f6dd.pkg ./BackgroundMusic-0.4.0-SNAPSHOT-b38f6dd
cd BackgroundMusic-0.4.0-SNAPSHOT-b38f6dd/Installer.pkg/
sudo cp -r Payload/Library/Audio/Plug-Ins/HAL/Background\ Music\ Device.driver /Library/Audio/Plug-Ins/HAL/
cp -r Payload/Applications/Background\ Music.app /Applications/
cd Scripts/

# This will probably print out some errors. The output explains them a bit.
sudo ./postinstall
echo $?
```

>Also, when will this be available in the normal version so tools like homebrew will download the universal binaries?

Releasing a new stable version is high on my to-do list, but I've been too busy for a while. Hopefully I'll get around to it soon. In the meantime, you should be able to use the prerelease (i.e. snapshot) cask:

```
brew tap homebrew/cask-versions
brew install --cask background-music-pre
```

**Comment by kyleneideck** [comment]
Thanks, it looks like this is the command that failed: `system_profiler SPAudioDataType | grep "Background Music"`

What do you get for `system_profiler SPAudioDataType`?

**Comment by dannymichel** [comment]
> Have you tried the latest snapshot build

yes but it says to install ``bgmxpchelper`` but doesn't say how/where to

**Comment by maximkrouk** [comment]
> yes but it says to install bgmxpchelper but doesn't say how/where to
The same problem 🥲

**Comment by ssbarnea** [comment]
OMG, I took me two hours to found out that this brilliant app is not silicon native. That is a big deal.

PS. Any change to enable sponsorship? I would rather use that than paying for a non open-source alternative.

**Comment by huyz** [comment]
Here's to wishing
